pkg/controller/servicebindingrequest/sbrcontroller.go

```diff
@@ -110,18 +110,30 @@ func updateFunc(logger *log.Log) func(updateEvent event.UpdateEvent) bool {
 		isConfigMap := isOfKind(e.ObjectNew, "ConfigMap")
 
 		if isSecret || isConfigMap {
-			dataFieldsAreEqual, err := compareObjectFields(e.ObjectNew, e.ObjectOld, "data")
+			dataFieldsAreEqual, err := compareObjectFields(e.ObjectOld, e.ObjectNew, "data")
 			if err != nil {
 				logger.Error(err, "error comparing object fields: %s", err.Error())
 				return false
 			}
-
 			logger.Debug("Predicate evaluated", "dataFieldsAreEqual", dataFieldsAreEqual)
 			return !dataFieldsAreEqual
 		}
 
-		// ignore updates to CR status in which case metadata.Generation does not change
-		return e.MetaOld.GetGeneration() != e.MetaNew.GetGeneration()
+		specsAreEqual, err := compareObjectFields(e.ObjectOld, e.ObjectNew, "spec")
+		if err != nil {
+			logger.Error(err, "")
+		}
+		logger.Debug("Predicate evaluated", "specsAreEqual", specsAreEqual)
```

Comment:
isn't it obvious - if there is not error displayed above, spec are equal? I would drop this statement from the log.

Reply:
The log was inspired from https://github.com/redhat-developer/service-binding-operator/blob/master/pkg/controller/servicebindingrequest/sbrcontroller.go#L215 . Should we take into the regard of refactoring that as well?

Reply:
Done https://github.com/redhat-developer/service-binding-operator/pull/597/commits/2899fd0b882ae30ee3b19d149d9329703c878e0c

Reply:
Done https://github.com/redhat-developer/service-binding-operator/pull/597/commits/b904a2186b93b06ef06213f2a23139572628abdc